### ISSUE TITLE
Brave considers https and wss as different origins when blocking cookies

### ISSUE THREAD

**Issue by STRML** [brave/browser-laptop#6641]
**Did you search for similar issues before submitting this one?**
Yes

**Describe the issue you encountered:**
This is intended to be a general ticket encompassing the underlying issue behind #4694 and #4918. There may be others.

The issue is that in the default mode, Brave considers a websocket connection on the same origin as "3rd party" and blocks sending cookies. This is unexpected behavior and breaks auth on many websockets.

**Expected behavior:**

For the purpose of blocking cookies, `http://mysite.com` and `ws://mysite.com` should be considered the same origin, as should `https://mysite.com` and `wss://mysite.com`.

- Platform (Win7, 8, 10? macOS? Linux distro?):
Any

- Brave Version (revision SHA):
0.12.15

- Steps to reproduce:
    1. Go to any site that does websocket authentication with cookies (for example, bitmex.com)
    2. Notice the websocket does not receive the `Cookie` header and auth fails.

- Any related issues:

Strangely, this appears to be triggered by a bug of some sort; on a fresh install I'm not able to replicate the Cookie header being blocked on `wss://` until a few cycles of logging in and out.

**Comment by luixxiul** [comment]
CC @diracdeltas for a thought

**Comment by STRML** [comment]
Working on a fix; this appears to be related to the devtools in some way, can replicate behavior on localhost but only with devtools open.

**Comment by STRML** [comment]
I was able to replicate this in `v0.12.15` but not in `master`.

In `master`, when the websocket upgrade request goes through [filtering](https://github.com/brave/browser-laptop/blob/1d3384a9186e451e64f72b5d8f6d9106b82cb82e/app/filtering.js#L254), `details` contains the following:

```json
{
  firstPartyUrl: 'http://localhost:2001/app/wallet',
  id: 79,
  method: 'GET',
  referrer: '',
  requestHeaders: {
    'Accept-Encoding': 'gzip, deflate',
    'Accept-Language': 'en-US',
    Connection: 'Upgrade',
    Cookie: 'connect.sid=s%3AkEYCYf7e_-k4-jTasWZiFFshUGv6i_FA.G%2F0FDMFLrfH46a7Wdj7tLJYAUPravxe6xIAw7d5QSQ0',
    Origin: 'http://localhost:2001',
    'Sec-WebSocket-Version': '13',
    Upgrade: 'websocket',
    'User-Agent': '...'
  },
  resourceType: 'other',
  tabId: -1,
  timestamp: 1484412030976.0388,
  url: 'ws://localhost:2001/realtime?heartbeat=1&_primuscb=LcTsW_-'
}
```

This runs through `getMainFrameUrl` which surprisingly returns the brave extension's url:

```
chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/_generated_background_page.html
```

Therefore, the cookie blocker is comparing `mnojpmjdmbbfmejpflffifhffcmidifd` to `localhost` which fails and the cookie is blocked.

In `master`, however, this is correctly tagged as an internal request and not blocked.

This was fixed in https://github.com/brave/browser-laptop/commit/dd0a066b928a4b73c87d908d8c1e8af7d64f6082.

Therefore I believe both #4694 and #4918 can be closed with a note that this is fixed in 0.13.
